src/app/model/PieceMovement.ts

```diff
@@ -0,0 +1,5 @@
+type Board = { squares: Square[] };
+
+interface PieceMovement {
```

Comment:
Dobry interfejs, właśnie o to chodzi z tą "uniwersalnością" :) 
Jedyne co, postaraj się, żeby position nie było stringiem. Bo to może być i równie dobrze: "Bo widzisz, to nasze wybory ukazują kim naprawdę jesteśmy o wiele bardziej niż nasze zdolności" :P 
Tutaj przydałby się jakiś typ / albo klasa / albo interfejs - skoro Ty tego nie implementujesz teraz (tak samo jak Board jest OK) który Ci np. daje już column i row, a nie musisz robić sobie potem wyciagania ze stringa.
Wtedy mozesz zalozyc, ze ten type Position zwroci Ci zawsze wlasciwa kolumne i wiersz, tak to nie wiesz co jest w stringu.

Reply:
(notka na jutro :-) )
Dobra, czyli chyba najlepiej by to wyglądało tak:
type Position = { column: Column, row: Row };
enum Column = {A,B,C,D,E,F,G,H};
enum Row = {1,2,3,4,5,6,7,8};

Reply:
Nie wiem, czy najlepiej, ale na pewno jakoś tak :) 
Hmmm... o ile jakiś typ z organiczonymi wartościami jak najbardziej ma tutaj sens, to zdaje mi się, że lepszym zastosowaniem niż enumy byłaby tutaj unia.
Nie zrobisz enuma, który jest liczbą (musi być nazwa), a to znacznie zmniejszyłoby tutaj czytelność.
Unia to np:
```
type Row = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8
type Column = 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H'
```

Reply:
No faktycznie lepiej to wygląda, ale nie mogę iterować po typie, przez co zdecydowałem się zrobić Board'a jako matrix :-)